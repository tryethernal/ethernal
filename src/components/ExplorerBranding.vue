<template>
    <v-card outlined :disabled="disabled">
        <v-form @submit.prevent="update()" v-model="valid">
            <v-card-text>
                <v-alert v-if="disabled" text type="warning">Upgrade your plan to activate branding customization.</v-alert>
                <v-alert v-if="successMessage" dense text type="success">{{ successMessage }}</v-alert>
                <v-alert v-if="errorMessage" dense text type="error">{{ errorMessage }}</v-alert>
                <v-row>
                    <v-col v-if="themes.light" cols="6">
                        <div v-for="(key, idx) in Object.keys(themes.light)" :key="idx">
                            <v-text-field
                                @focus="selectedColorPicker = key"
                                @blur="selectedColorPicker = null"
                                outlined
                                dense
                                v-model="themes.light[key]"
                                :label="key.charAt(0).toUpperCase() + key.slice(1) + ' Color'">
                                <template v-slot:prepend>
                                    <v-icon @click="selectedColorPicker = selectedColorPicker ? null : key" :color="themes.light[key]">mdi-square</v-icon>
                                </template>
                            </v-text-field>
                            <v-color-picker
                                v-show="selectedColorPicker == key"
                                v-model="themes.light[key]"
                                dot-size="25"
                                hide-inputs></v-color-picker>
                        </div>
                    </v-col>
                    <v-col cols="6">
                        <v-text-field
                            dense
                            outlined
                            v-model="themes.logo"
                            label="Logo URL">
                            <template v-slot:prepend v-if="themes.logo">
                                <v-img :src="themes.logo" max-width="150" class="mb-4"></v-img>
                            </template>
                        </v-text-field>
                        <v-text-field
                            dense
                            outlined
                            v-model="themes.favicon"
                            label="Favicon URL">
                            <template v-slot:prepend v-if="themes.favicon">
                                <v-img :src="themes.favicon" class="mb-4"></v-img>
                            </template>
                        </v-text-field>
                        <v-autocomplete outlined dense append-icon=""
                            label="Font"
                            :hint="'Font needs to be available on Google Fonts (default is Roboto)'"
                            v-model="themes.font"
                            :items="fonts"
                            :loading="fontSearchLoading"
                            :search-input.sync="queryFont"
                            hide-no-data
                            persistent-hint
                            no-filter></v-autocomplete>
                        <v-text-field
                            dense
                            outlined
                            v-model="themes.banner"
                            label="Banner Text"></v-text-field>
                        <h4 class="mb-2">Links</h4>

                        <New-Explorer-Link v-for="(link, idx) in themes.links" :name="link.name" :url="link.url" :icon="link.icon" :index="idx" :uid="link.uid" :key="link.uid" @updated="onUpdatedExplorerLink" @removed="onRemovedExplorerLink" />

                        <a @click.stop="addExplorerLink()">Add</a>
                    </v-col>
                </v-row>
                <v-card-actions>
                    <v-spacer></v-spacer>
                    <v-btn :loading="loading" color="primary" :disabled="!valid" type="submit">Update</v-btn>
                </v-card-actions>
            </v-card-text>
        </v-form>
    </v-card>
</template>

<script>
import NewExplorerLink from './NewExplorerLink.vue';

export default {
    name: 'ExplorerBranding',
    props: ['explorer', 'disabled'],
    components: {
        NewExplorerLink,
    },
    data: () => ({
        successMessage: null,
        errorMessage: null,
        explorerLinks: [],
        queryFont: null,
        fonts: [],
        fontSearchLoading: false,
        selectedColorPicker: null,
        valid: false,
        loading: false,
        themes: {
            links: []
        }
    }),
    mounted() {
        this.themes = { ...this.explorer.themes, ...this.themes };
        this.themes.light = { ...this.$vuetify.theme.themes.light, ...this.explorer.themes.light };

        if (this.themes.font)
            this.fonts.push(this.themes.font);
        if (this.explorer.themes.links)
            this.themes.links = this.explorer.themes.links;
    },
    methods: {
        onUpdatedExplorerLink(data) {
            this.themes.links[data.index] = data.link;
        },
        onRemovedExplorerLink(uid) {
            const index = this.themes.links.map(l => l.uid).indexOf(uid);
            if (index >= 0)
                this.themes.links.splice(index, 1);
        },
        addExplorerLink() {
            this.themes.links.push({ url: null, name: null, icon: null, uid: Math.floor(Math.random() * 10000) });
        },
        update() {
            this.loading = true;
            this.successMessage = null;
            this.errorMessage = null;
            this.server.updateExplorerBranding(this.explorer.id, this.themes)
                .then(() => {
                    this.successMessage = 'Branding updated successfully.';
                })
                .catch(error => {
                    this.errorMessage = error.response && error.response.data || 'Error while updating branding. Please retry.';
                })
                .finally(() => {
                    this.loading = false;
                });
        },
    },
    watch: {
        queryFont(query) {
            if (!query || query == this.themes.font) return;

            this.fontSearchLoading = true;
            this.server.searchFont(this.queryFont)
                .then(({ data }) => this.fonts = data)
                .catch(console.log)
                .finally(() => this.fontSearchLoading = false);
        }
    }
}
</script>
