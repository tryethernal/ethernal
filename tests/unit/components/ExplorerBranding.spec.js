import flushPromises from 'flush-promises';
import MockHelper from '../MockHelper';

import ExplorerBranding from '@/components/ExplorerBranding.vue';

let helper;
beforeEach(() => {
    jest.clearAllMocks()
    helper = new MockHelper();
});

describe('ExplorerBranding.vue', () => {
    it('Should display branding options', async () => {
        const wrapper = helper.mountFn(ExplorerBranding, {
            stubs: ['New-Explorer-Link', 'v-color-picker'],
            propsData: {
                disabled: false,
                explorer: {
                    id: 1,
                    themes: {
                        links: [{ icon: 'twitter', url: 'twitter.com', name: 'twitter', uid: 1 }],
                        light: {
                            primary: '#32a852'
                        }
                    }
                }
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should display upgrade message', async () => {
        const wrapper = helper.mountFn(ExplorerBranding, {
            stubs: ['New-Explorer-Link', 'v-color-picker'],
            propsData: {
                disabled: true,
                explorer: {
                    id: 1,
                    themes: {
                        links: [{ icon: 'twitter', url: 'twitter.com', name: 'twitter', uid: 1 }],
                        light: {
                            primary: '#32a852'
                        }
                    }
                }
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
