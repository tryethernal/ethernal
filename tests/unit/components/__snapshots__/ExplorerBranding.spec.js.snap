// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplorerBranding.vue Should display branding options 1`] = `
<div class="v-card v-sheet v-sheet--outlined theme--light">
  <form novalidate="novalidate" class="v-form">
    <div class="v-card__text">
      <!---->
      <!---->
      <!---->
      <div class="row">
        <div class="col col-6">
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(50, 168, 82); caret-color: #32a852;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-21" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Primary Color</label><input id="input-21" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#32a852" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(66, 66, 66); caret-color: #424242;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-26" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Secondary Color</label><input id="input-26" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#424242" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(130, 177, 255); caret-color: #82B1FF;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-31" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Accent Color</label><input id="input-31" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#82B1FF" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(255, 82, 82); caret-color: #FF5252;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-36" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Error Color</label><input id="input-36" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#FF5252" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(33, 150, 243); caret-color: #2196F3;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-41" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Info Color</label><input id="input-41" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#2196F3" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(76, 175, 80); caret-color: #4CAF50;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-46" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Success Color</label><input id="input-46" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#4CAF50" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(251, 140, 0); caret-color: #FB8C00;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-51" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Warning Color</label><input id="input-51" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#FB8C00" width="300" style="display: none;"></v-color-picker-stub>
          </div>
        </div>
        <div class="col col-6">
          <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
            <div class="v-input__control">
              <div class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-text-field__slot"><label for="input-6" class="v-label theme--light" style="left: 0px; position: absolute;">Logo URL</label><input id="input-6" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
            <div class="v-input__control">
              <div class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-text-field__slot"><label for="input-9" class="v-label theme--light" style="left: 0px; position: absolute;">Favicon URL</label><input id="input-9" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
            <div class="v-input__control">
              <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-12" class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-select__slot"><label for="input-12" class="v-label theme--light" style="left: 0px; position: absolute;">Font</label><input id="input-12" type="text" autocomplete="off"><input type="hidden"></div>
                <div class="v-menu">
                  <!---->
                </div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper">
                    <div class="v-messages__message">Font needs to be available on Google Fonts (default is Roboto)</div>
                  </transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
            <div class="v-input__control">
              <div class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-text-field__slot"><label for="input-16" class="v-label theme--light" style="left: 0px; position: absolute;">Banner Text</label><input id="input-16" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <h4 class="mb-2">Links</h4>
          <new-explorer-link-stub uid="1" index="0" icon="twitter" url="twitter.com" name="twitter"></new-explorer-link-stub> <a>Add</a>
        </div>
      </div>
      <div class="v-card__actions">
        <div class="spacer"></div> <button type="submit" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">Update</span></button>
      </div>
    </div>
  </form>
</div>
`;

exports[`ExplorerBranding.vue Should display upgrade message 1`] = `
<div class="v-card v-card--disabled v-sheet v-sheet--outlined theme--light">
  <form novalidate="novalidate" class="v-form">
    <div class="v-card__text">
      <div role="alert" class="v-alert v-sheet theme--dark v-alert--text warning--text">
        <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-exclamation theme--dark warning--text"></i>
          <div class="v-alert__content">Upgrade your plan to activate branding customization.</div>
        </div>
      </div>
      <!---->
      <!---->
      <div class="row">
        <div class="col col-6">
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(50, 168, 82); caret-color: #32a852;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-79" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Primary Color</label><input id="input-79" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#32a852" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(66, 66, 66); caret-color: #424242;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-84" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Secondary Color</label><input id="input-84" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#424242" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(130, 177, 255); caret-color: #82B1FF;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-89" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Accent Color</label><input id="input-89" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#82B1FF" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(255, 82, 82); caret-color: #FF5252;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-94" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Error Color</label><input id="input-94" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#FF5252" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(33, 150, 243); caret-color: #2196F3;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-99" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Info Color</label><input id="input-99" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#2196F3" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(76, 175, 80); caret-color: #4CAF50;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-104" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Success Color</label><input id="input-104" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#4CAF50" width="300" style="display: none;"></v-color-picker-stub>
          </div>
          <div>
            <div class="v-input v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
              <div class="v-input__prepend-outer"><button type="button" class="v-icon notranslate v-icon--link mdi mdi-square theme--light" style="color: rgb(251, 140, 0); caret-color: #FB8C00;"></button></div>
              <div class="v-input__control">
                <div class="v-input__slot">
                  <fieldset aria-hidden="true">
                    <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                  </fieldset>
                  <div class="v-text-field__slot"><label for="input-109" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Warning Color</label><input id="input-109" type="text"></div>
                </div>
                <div class="v-text-field__details">
                  <div class="v-messages theme--light">
                    <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
            <v-color-picker-stub canvasheight="150" dotsize="25" hideinputs="true" mode="rgba" swatchesmaxheight="150" value="#FB8C00" width="300" style="display: none;"></v-color-picker-stub>
          </div>
        </div>
        <div class="col col-6">
          <div class="v-input v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
            <div class="v-input__control">
              <div class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-text-field__slot"><label for="input-64" class="v-label theme--light" style="left: 0px; position: absolute;">Logo URL</label><input id="input-64" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-input v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
            <div class="v-input__control">
              <div class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-text-field__slot"><label for="input-67" class="v-label theme--light" style="left: 0px; position: absolute;">Favicon URL</label><input id="input-67" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-input v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
            <div class="v-input__control">
              <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-70" class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-select__slot"><label for="input-70" class="v-label theme--light" style="left: 0px; position: absolute;">Font</label><input id="input-70" type="text" autocomplete="off"><input type="hidden"></div>
                <div class="v-menu">
                  <!---->
                </div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper">
                    <div class="v-messages__message">Font needs to be available on Google Fonts (default is Roboto)</div>
                  </transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-input v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined">
            <div class="v-input__control">
              <div class="v-input__slot">
                <fieldset aria-hidden="true">
                  <legend style="width: 0px;"><span class="notranslate">​</span></legend>
                </fieldset>
                <div class="v-text-field__slot"><label for="input-74" class="v-label theme--light" style="left: 0px; position: absolute;">Banner Text</label><input id="input-74" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <h4 class="mb-2">Links</h4>
          <new-explorer-link-stub uid="1" index="0" icon="twitter" url="twitter.com" name="twitter"></new-explorer-link-stub> <a>Add</a>
        </div>
      </div>
      <div class="v-card__actions">
        <div class="spacer"></div> <button type="submit" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">Update</span></button>
      </div>
    </div>
  </form>
</div>
`;
